### PR TITLE
[WIP] added push_datapackage and pull_datapackage

### PR DESCRIPTION
**Pull request by roll**
Work in progress will wait a discussion and https://github.com/frictionlessdata/testsuite-py tests added and passed.

### CODE REVIEW

**Comment by roll** [comment]
@pwalsh @vitorbaptista 
Please take a look.

It works with 9 datasets (push/pull to SQL/BigQuery) from `testsuite-py`:
https://travis-ci.org/frictionlessdata/testsuite-py/jobs/119227783

**Comment by pwalsh** [comment]
@roll :8ball:

**Comment by pwalsh** [comment]
@roll also, really cool using `behave` for the integration tests!

**Comment by pwalsh** [comment]
@roll please get final signoff on this from @vitorbaptista

File: README.md [hunk 1]
```diff
@@ -110,6 +110,41 @@ schema = registry.get('tabular')  # Change to your schema ID
...
+
+> All parameters should be used as keyword arguments.
+
+```python
+from datapackage import push_datapackage, pull_datapackage
+
+# Push
+push_datapackage(
+    descriptor='descriptor_path',
+    backend='mystorage, '**<mystorage_options>)
```
Comment (vitorbaptista) [inline README.md]: The quote is in the wrong place

File: datapackage/mappers.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    """
+    schemas = deepcopy(schemas)
+    for schema in schemas:
+        for fk in schema.get('foreignKeys', []):
+            resource = fk['reference']['resource']
+            if resource != 'self':
+                if resource not in mapping:
+                    message = 'Not resource "%s" for foreign key "%s"'
+                    message = message % (resource, fk)
+                    raise ValueError(message)
```
Comment (vitorbaptista) [inline datapackage/mappers.py]: Add this exception to the docstring

**Comment by vitorbaptista** [comment]
@roll The code looks good, it just needs a few tweaks to be more consistent with the overall style of this package. For instance, instead of having `pull_datapackage` and `push_datapackage`, we'd have `DataPackage.pull()` as a class method which returns a `DataPackage` instance, and `datapackage.push()` as an instance method.

To be honest, I prefer `DataPackage.from_storage()` and `datapackage.to_storage()`, but that might be because "push" is a tricky word for me. It sounds the same as the Portuguese word "puxe", which means "pull". It takes me a second to translate, which makes pulling doors with "push" signs in them :sweat_smile:

Push/pull are very common for devs, though, so I'm happy to keep it.

**Comment by roll** [comment]
@vitorbaptista 
It's following of our prev discussion - https://github.com/frictionlessdata/jsontableschema-py/pull/51#issuecomment-194185012

I think we decided for now to have temp solution:
```
- push_resource (filesystem -> db)
- pull_resource (db -> filesystem)
- push_datapackage (filesystem -> db)
- pull_datapackage (db -> filesystem)
```

So I've intentionally done it without touching `DataPackage` API and OOP-approach. I think we could integrate it with the main API after `jsontableschm.Resource` will be designed and deprecate API from `pushpull.py`.

cc @pwalsh

File: .venv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Activate: source .venv
+virtualenv .python
+source .python/bin/activate
+pip install --upgrade -e .
+pip install tox
```
Comment (vitorbaptista) [inline .venv]: Is this a common pattern? This feels like it should belong in the README instead, maybe in a "Running the tests" subsection inside "Developer notes".

Comment (roll) [inline .venv]: Ohh my bad system `.gitignore` was broken

**Comment by vitorbaptista** [comment]
You're right. Just check a few typos on the README (like `exmples`) and feel free to merge :+1:

**Comment by roll** [comment]
Great!

- typos fixed 
- tested against `testsuite-py` again
- merged

**Pull request merged**